synapse/tests/test_lib_json.py

```diff
@@ -0,0 +1,160 @@
+        self.eq(outval + b'\n', s_json.dumps(inval, newline=True))
+
+        buf = io.BytesIO()
+        s_json.dump(inval, buf)
+        self.eq(outval, buf.getvalue())
+
+    async def test_lib_json_dumps(self):
+        self.eq(b'{"c":"d","a":"b"}', s_json.dumps({'c': 'd', 'a': 'b'}))
+        self.eq(b'{"a":"b","c":"d"}', s_json.dumps({'c': 'd', 'a': 'b'}, sort_keys=True))
+        self.eq(b'{\n  "c": "d",\n  "a": "b"\n}', s_json.dumps({'c': 'd', 'a': 'b'}, indent=True))
+        self.eq(b'{"c":"d","a":"b"}\n', s_json.dumps({'c': 'd', 'a': 'b'}, newline=True))
+
+        with self.raises(s_exc.MustBeJsonSafe) as exc:
+            s_json.dumps({}.items())
+        self.eq(exc.exception.get('mesg'), 'Type is not JSON serializable: dict_items')
+
+        with self.raises(s_exc.MustBeJsonSafe) as exc:
+            s_json.dumps({1: 'foo'})
+        self.eq(exc.exception.get('mesg'), 'Dict key must be str')
+
+        with self.raises(s_exc.MustBeJsonSafe) as exc:
+            s_json.dumps({'\ud83d\ude47': {}.items()})
+        self.eq(exc.exception.get('mesg'), 'Object of type dict_items is not JSON serializable')
+
+        self.eq(b'"dict_items([])"', s_json.dumps({}.items(), default=str))
+
+    async def test_lib_json_dump(self):
+        with self.getTestDir() as dirn:
+            binfn = s_common.genpath(dirn, 'bin.json')
+
+            with open(binfn, 'wb') as binfp:
+                s_json.dump({'c': 'd', 'a': 'b'}, binfp)
+
+            with open(binfn, 'rb') as binfp:
+                self.eq(b'{"c":"d","a":"b"}', binfp.read())
+
+        buf = io.BytesIO()
+        s_json.dump({'c': 'd', 'a': 'b'}, buf)
+        self.eq(b'{"c":"d","a":"b"}', buf.getvalue())
+
+    async def test_jsload(self):
+        with self.getTestDir() as dirn:
+            with s_common.genfile(dirn, 'jsload.json') as fp:
+                fp.write(b'{"a":"b"}')
+
+            obj = s_json.jsload(dirn, 'jsload.json')
+            self.eq({'a': 'b'}, obj)
+
+            s_common.genfile(dirn, 'empty.json').close()
+            self.none(s_json.jsload(dirn, 'empty.json'))
+
+    async def test_jslines(self):
+        with self.getTestDir() as dirn:
+            with s_common.genfile(dirn, 'jslines.json') as fp:
+                fp.write(b'{"a":"b"}\n{"c":"d"}')
+
+            objs = [k for k in s_json.jslines(dirn, 'jslines.json')]
+            self.len(2, objs)
+            self.eq([{'a': 'b'}, {'c': 'd'}], objs)
+
+    async def test_jssave(self):
+        with self.getTestDir() as dirn:
+            s_json.jssave({'a': 'b'}, dirn, 'jssave.json')
+
+            with s_common.genfile(dirn, 'jssave.json') as fd:
+                data = fd.read()
+
+            self.eq(data, b'{\n  "a": "b"\n}')
+
+    async def test_lib_json_reqjsonsafe(self):
+        self.none(s_json.reqjsonsafe('foo'))
+        self.none(s_json.reqjsonsafe({'foo': 'bar'}))
+        self.none(s_json.reqjsonsafe(['foo', 'bar']))
+
+        with open('/dev/null', 'rb') as fp:
```

Comment:
Can we generate a temp directory and open a file in that instead of using `/dev/null` ?

Reply:
obe, I changed it to a io.BytesIO()